pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -38,22 +38,35 @@ type PredicateMetadataFactory struct {
 	podLister algorithm.PodLister
 }
 
+// AntiAffinityTerm's topology key value used in predicate metadata
+type topologyPair struct {
+	key   string
+	value string
+}
+
 //  Note that predicateMetadata and matchingPodAntiAffinityTerm need to be declared in the same file
 //  due to the way declarations are processed in predicate declaration unit tests.
 type matchingPodAntiAffinityTerm struct {
 	term *v1.PodAffinityTerm
 	node *v1.Node
 }
 
+// topologyPairsMaps keeps topologyPairToAntiAffinityPods and antiAffinityPodToTopologyPairs in sync
+// as they are the inverse of each others.
+type topologyPairsMaps struct {
+	topologyPairToAntiAffinityPods map[topologyPair][]*v1.Pod
+	antiAffinityPodToTopologyPairs map[string][]topologyPair
```

Comment:
Did you guys see this comment :point_up:

Reply:
@bsalamat The second map is to provide faster deletion of pods from the metadata. Having only the deleted pod, we would have to check all the topology pairs in order to delete the pod from the metadata. Do you think there is a better way to do it?. If I understand correctly, In case of converting the first map to the structure mentioned above, we would still need to check all pairs in order to delete a pod from the metadata.

Reply:
It makes sense. Thanks for reminding me.